pandas/core/frame.py

```diff
@@ -1943,6 +1944,9 @@ def _from_arrays(
         -------
         DataFrame
         """
+        if dtype is not None:
+            dtype = pandas_dtype(dtype)
```

Comment:
can the dtype parameter of pandas_dtype be typed?

Reply:
Typing it as "Optional[Dtype]" leads to

pandas/core/dtypes/common.py:1758: error: Argument 1 to "find" of "Registry" has incompatible type "Union[Any, str, Type[str], Type[complex], None]"; expected "Union[Type[ExtensionDtype], str]"